db/migrate/20210414203512_convert_primary_language_to_iso_code.rb

```diff
@@ -0,0 +1,124 @@
+    end
+
+    existing_primary_languages = Patient.where(purged: false).distinct.pluck(:primary_language)
+    existing_primary_languages.reject!(&:blank?)
+    existing_secondary_languages = Patient.where(purged: false).distinct.pluck(:secondary_language)
+    existing_secondary_languages.reject!(&:blank?)
+
+    # Create an translation in automatic_translations, unless we already have it defined in `manual_translations`
+    existing_primary_languages.each do |el|
+      automatic_translations[el.to_sym] = (match_language(el) || 'eng').to_s if (manual_translations[el.to_sym].nil?)
+    end
+    # Create an translation in automatic_translations, unless we already have it defined in `manual_translations`
+    existing_secondary_languages.each do |el|
+      automatic_translations[el.to_sym] = (match_language(el) || 'eng').to_s if (manual_translations[el.to_sym].nil?)
+    end
+
+    automatic_translations.each do |key, value|
+      # Patient records updated here are solely updated (no History Items included)
+      Patient.where(purged: false).where(primary_language: key).update_all(primary_language: value)
+      Patient.where(purged: false).where(secondary_language: key).update_all(secondary_language: value)
+    end
+    custom_translations.each do |key, value|
+      # Patient records updated here WILL have History Items included for them.
+      ids = Patient.where(purged: false).where(primary_language: key.to_s).distinct.pluck(:id)
+      ids.each do |id|
+        note = "Primary language was listed as '#{key}' \
+        which could not be matched to the standard language list. This monitoree’s primary \
+        language has been updated to '#{PATIENT_HELPER_FILES[:languages][value.to_sym][:display]}'. You \
+        may update this value."
+        History.create!(patient_id: id,
+                      created_by: "Sara Alert System",
+                      comment: note,
+                      history_type: 'System Note'
+                      )
+      end
+      Patient.where(purged: false).where(primary_language: key).update_all(primary_language: value)
+
+      ids = Patient.where(purged: false).where(secondary_language: key).distinct.pluck(:id)
+      ids.each do |id|
+        note = "Secondary language was listed as '#{key}' \
+        which could not be matched to the standard language list. This monitoree’s primary \
+        language has been updated to '#{PATIENT_HELPER_FILES[:languages][value.to_sym][:display]}'. You \
+        may update this value."
+        History.create!(patient_id: id,
+                      created_by: "Sara Alert System",
+                      comment: note,
+                      history_type: 'System Note'
+                      )
+      end
+      Patient.where(purged: false).where(secondary_language: key).update_all(secondary_language: value)
+    end
+  end
+
+
+  def down
+    execute <<-SQL.squish
+      UPDATE patients SET primary_language = legacy_primary_language
```

Comment:
might be more performant to set both of them in the same query

Reply:
Combined the two queries.